tests/Processor/SanitizeDataProcessorTest.php

```diff
@@ -193,6 +192,36 @@ public function processDataProvider()
                     ],
                 ],
             ],
+            [
+                [
+                    'extra_context' => [
```

Comment:
Please add test with depth 2 or 3. So we could see the method are working with deep depth.